src/Open3D/Visualization/Rendering/Filament/FilamentGeometryBuffersBuilder.cpp

```diff
@@ -188,6 +180,89 @@ std::tuple<VertexBufferHandle, IndexBufferHandle> PointCloudBuffersBuilder::Cons
+    auto& engine = EngineInstance::GetInstance();
+    auto& resourceManager = EngineInstance::GetResourceManager();
+
+    const size_t nVertices = geometry_.points_.size();
+
+    // Converting vertex normals to float base
+    std::vector<Eigen::Vector3f> normals;
+    normals.resize(nVertices);
+    for (size_t i = 0; i < nVertices; ++i) {
+        normals[i] = geometry_.normals_[i].cast<float>();
+    }
+
+    const size_t normalsBytesCount = nVertices * 4 * sizeof(float);
+    auto* float4Normals = static_cast<math::float4*>(malloc(normalsBytesCount));
+    for (size_t i = 0; i < nVertices; ++i) {
+        auto v = normals[i];
+        float4Normals[i] = {v(0), v(1), v(2), 0.f};
+    }
+
+    const auto vBytesCount = nVertices * sizeof(ColoredVertex);
+    auto* vBytes = malloc(vBytesCount);
+
+    auto plainVertices = static_cast<ColoredVertex*>(vBytes);
+    for (size_t i = 0; i < nVertices; ++i) {
+        ColoredVertex& element = plainVertices[i];
+
+        element.SetVertexPosition(geometry_.points_[i]);
+        element.color = float4Normals[i];
+    }
+
+    free(float4Normals);
+
+    size_t stride = sizeof(ColoredVertex);
+
+    VertexBuffer* vbuf =
+            VertexBuffer::Builder()
+                    .bufferCount(1)
+                    .vertexCount(nVertices)
+                    .attribute(VertexAttribute::POSITION, 0,
+                               VertexBuffer::AttributeType::FLOAT3,
+                               0, stride)
+                    .attribute(VertexAttribute::COLOR, 0,
+                               VertexBuffer::AttributeType::FLOAT4,
+                               offsetof(ColoredVertex, color), stride)
+                    .build(engine);
+
+    VertexBufferHandle vbHandle;
+    if (vbuf) {
+        vbHandle = resourceManager.AddVertexBuffer(vbuf);
+    } else {
+        free(vBytes);
+
+        return {};
+    }
+
+    VertexBuffer::BufferDescriptor vertexbufferDescriptor(vBytes, vBytesCount);
+    vertexbufferDescriptor.setCallback(GeometryBuffersBuilder::DeallocateBuffer);
+    vbuf->setBufferAt(engine, 0, std::move(vertexbufferDescriptor));
+
+    const auto iBytesCount = nVertices*sizeof(IndexType);
+    auto *iBytes = static_cast<IndexType*>(malloc(iBytesCount));
+    for (std::uint32_t i = 0; i < nVertices; ++i) {
+        iBytes[i] = i;
+    }
+
+    auto ibHandle = resourceManager.CreateIndexBuffer(nVertices, sizeof(IndexType));
+    if (!ibHandle) {
+        free(iBytes);
+        return {};
+    }
+
+    auto ibuf = resourceManager.GetIndexBuffer(ibHandle).lock();
+
+    // Moving copied indices to IndexBuffer
+    // they will be freed later with freeBufferDescriptor
+    IndexBuffer::BufferDescriptor indicesDescriptor(iBytes, iBytesCount);
+    indicesDescriptor.setCallback(GeometryBuffersBuilder::DeallocateBuffer);
+    ibuf->setBuffer(engine, std::move(indicesDescriptor));
+
+    return std::make_tuple(vbHandle, ibHandle);
+}
+
 filament::Box PointCloudBuffersBuilder::ComputeAABB() {
```

Comment:
If I understand CreateNormalsGhost() correctly, we are creating geometry and assigning the normals to the color attribute. Why do we need a new geometry? Can't we use the buffers we create for usual rendering and do the Filament equivalent of `color = vec4(normals.xyz, 1.0);` in the shader? (hopefully I got that GLSL right)

Reply:
Yeah, you understand it right. So, why do we need a geometry? Problem is, that Filament optimizes out tangent buffer access from unlit materials, and for normals rendering purpose we can't use lit material. That's why we need another instance of geometry.
As I said in comment above, we can't just use value from tangents, because it is quaternion) There are some helper functions to get from this quaternion a normal, but guess what?)) They optimized out from unlit shader.

Reply:
Hmm, but, I think, I just figured out some kind of hacky solution.
I'll put tangents not in TANGENTS attribute, but instead I could try to put them into CUSTOM0 attribute and then, they would be optimized out, maybe)
But, I'm afraid that it will bork the lit material. Will check, anyway.

Reply:
Yeap, it works)